extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js

```diff
@@ -87,5 +84,37 @@ require([
 					scrollFix.restoreScrollY($globalNavigation);
 				});
 		}
+	}
+
+	$(function () {
+		var $entryPoint, $authEntryPoints;
+
+		$entryPoint = $('#AccountNavigation');
+
+		if (!win.wgUserName && $entryPoint.hasClass('newAuth')) {
+			$authEntryPoints = $entryPoint.find('.auth-link.register, .auth-link.sign-in, a.sign-in');
+
+			$authEntryPoints.click(function (event) {
+				//Prevent opening modal with shift / alt / ctrl / let only left mouse click
+				if (event.which !== 1 || event.shiftKey || event.altKey || event.metaKey || event.ctrlKey) {
+					return;
+				}
+				if (event) {
+					event.preventDefault();
+					event.stopPropagation();
+				}
+
+				require(['AuthModal'], function (authModal) {
+					if (event.target.classList.contains('register')) {
+						authModal.register();
+					} else {
+						authModal.login();
+					}
+				});
+			});
```

Comment:
what do you think about extracting code responsible for new modal to separate function?